SCAutolib/models/user.py

```diff
@@ -0,0 +1,153 @@
+from pathlib import Path
+from python_freeipa.client_meta import ClientMeta
+
+from SCAutolib import run, logger
+
+
+class User:
+    """
+    Generic class to represent system users.
+    """
+
+    def __init__(self, username: str, password: str, pin: str, card = None,
+                 cnf: Path = None, key: Path = None, cert: Path = None):
+
+        """
+        :param username: Username for the system user
+        :type: str
+        :param password: Password for the system user
+        :type str
+        :param pin: Smart card pin for the system user
+        :type str
+        :param card: Card to be associated with the user
+        :type Card
+        :param cnf: CNF file to be associated with the user
+        :type Path
+        :param key: Key to be assoicated with the user
+        :type Path
+        :param cert: Certificate to be associated with the user.
+        :type Path
+        """
+
+        self.username = username
+        self.password = password
+        self.pin = pin
+        self._card = card
```

Comment:
Here we would have recursive dependencies on the `VirtualCard` because this type of card for enrollment requires a user. So I would suggest moving this setter to the property and do not have it in the constructor. In this way steps for creating a user with a virtual smart card would be more obvious: create user -> create a card and pass the user to it -> enroll card (because enrollment requires a user name and pin) or whatever -> set a card for the user via setter

Reply:
Oh, sorry. You already have a setter for the card. So, just remove it from here